### PR TITLE
Added subject cards with grid layout. Implemented bundles for Parse..

### PR DESCRIPTION
**Pull request by SSGrady**
.. Sign Up.

Demo:
cardview_and_bundlesPR_demo

### CODE REVIEW

File: app/src/main/java/com/example/grutor/Fragments/HomeFragment.java [hunk 1]
```diff
@@ -47,8 +55,28 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
...
         btnlogOut = view.findViewById(R.id.btnlogOut);
+        // Recycler View population.
+        rvSubjects = view.findViewById(R.id.rvSubjects);
+        titles = new ArrayList<>();
+        images = new ArrayList<>();
+
+        titles.add("First Subject");
+        titles.add("Second Subject");
+        titles.add("Third Subject");
+        titles.add("Fourth Subject");
```
Comment (busyjun) [inline app/src/main/java/com/example/grutor/Fragments/HomeFragment.java]: Put the string into strings.xml

File: app/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -50,6 +40,14 @@
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintTop_toTopOf="@+id/tvWelcomeUser" />
 
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/rvSubjects"
+            android:layout_width="409dp"
+            android:layout_height="400dp"
```
Comment (busyjun) [inline app/src/main/res/layout/fragment_home.xml]: layout should match parent and add layout margin. you can change the screen size and check if UI is scalable.

File: app/src/main/java/com/example/grutor/Activites/RegisterActivity.java [hunk 3]
```diff
@@ -58,20 +59,21 @@ public void onClick(View v) {
...
             public void registerUser() {
                 // Create the ParseUser
-                user = new ParseUser();
+               // user = new ParseUser();
                 // Set core properties
                 Log.i(TAG, "Inside the registerUser function.");
                 String email = etUserLogin.getText().toString();
                 String password = etPassword.getText().toString();
                 String firstName = etFirstName.getText().toString();
+                String fullName = etFirstName.getText().toString() + " " + etLastName.getText().toString();
```
Comment (busyjun) [inline app/src/main/java/com/example/grutor/Activites/RegisterActivity.java]: you should check the format of email. the length of user name and password. [Optional] check if any duplicated user name too( or show a error message when user sign up a duplicated user name).

**Review by busyjun — APPROVED** [review]

File: app/src/main/java/com/example/grutor/Fragments/HomeFragment.java [hunk 4]
```diff
@@ -60,15 +59,19 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
...
-        images.add(R.drawable.icons8_chat_bubble_50);
-        images.add(R.drawable.icons8_chat_bubble_51);
-        images.add(R.drawable.user_filled_24);
-        images.add(R.drawable.user_outline_24);
+        titles.add("Math");
+        titles.add("English");
+        titles.add("Science");
+        titles.add("History");
+        titles.add("Government");
+        titles.add("Economics");
```
Comment (busyjun) [inline app/src/main/java/com/example/grutor/Fragments/HomeFragment.java]: put strings into resource.

File: app/src/main/java/com/example/grutor/Fragments/HomeFragment.java [hunk 5]
```diff
@@ -31,13 +31,12 @@ public class HomeFragment extends Fragment {
...
     public Button btnlogOut;
-    public ParseUser currentUser;
-    public static final String KEY_PARSE_USER_NAME = "name";
+    protected ParseUser currentUser;
     protected ParseUser user;
-    public static String welcomeMessage = "";
-    RecyclerView rvSubjects;
-    List<String> titles;
-    List<Integer> images;
+    private static String welcomeMessage = "";
```
Comment (busyjun) [inline app/src/main/java/com/example/grutor/Fragments/HomeFragment.java]: why need to define welcomeMessage to be static?

**Review by busyjun — APPROVED** [review]

File: app/src/main/java/com/example/grutor/Activites/RegisterActivity.java (hunk 3 above)
Comment (SSGrady) [reply app/src/main/java/com/example/grutor/Activites/RegisterActivity.java]: I have implemented a new email checker function and the changes about adding and extracting String resources from strings.xml. I will go back to checking the length of the user name and password later. I added a todo, thanks!

**Review by busyjun — APPROVED** [review]

File: app/src/main/java/com/example/grutor/Adapters/SubjectAdapter.java [hunk 6]
```diff
@@ -22,11 +24,13 @@ public class SubjectAdapter extends RecyclerView.Adapter<SubjectAdapter.ViewHold
     List<String> titles;
     List<Integer> images;
     LayoutInflater inflater;
+    Context context;
```
Comment (busyjun) [inline app/src/main/java/com/example/grutor/Adapters/SubjectAdapter.java:27]: define to private

**Review by busyjun — APPROVED** [review]

**Pull request merged**